### PR TITLE
Make barrier option to automatically destroy the barrier if jobs belonging to it fails

### PR DESCRIPTION
**Pull request by mudler**
See related Progress issue: https://progress.opensuse.org/issues/27409

Needs: https://github.com/os-autoinst/os-autoinst/pull/880

### CODE REVIEW

File: lib/OpenQA/Resource/Locks.pm [hunk 1]
```diff
@@ -125,15 +128,24 @@ sub barrier_create {
...
+    my ($name, $jobid, $where, $check_dead_job) = @_;
     return -1 unless $name && $jobid;
     my $barrier = _get_lock($name, $jobid, $where);
     return -1 unless $barrier;
+    my $jobschema = OpenQA::ResourceAllocator->instance->schema()->resultset("Jobs");
     my @jobs = split(/,/, $barrier->locked_by // '');
+
+    do { $barrier->delete; return -1 }
+      if $check_dead_job
+      && grep { $final_states{$_} }
```
Comment (foursixnine) [inline lib/OpenQA/Resource/Locks.pm]: Just a cosmetic change, I'd add a line here.

File: t/13-joblocks.t [hunk 2]
```diff
@@ -235,4 +235,91 @@ ok($res, 'barrier1 is still there');
...
+    $t->post_ok($b_prefix . '/barrier2', form => {action => 'wait'})->status_is(410);
+    set_token_header($t->ua, 'token' . $jB);
+    $t->post_ok($b_prefix . '/barrier2', form => {action => 'wait'})->status_is(410);
+    return 1;
+}
+
+test_barrier_destroy($_, "jA")
+  && test_barrier_destroy($_, "jB")
+  && test_barrier_destroy($_, "jC")
+  for (OpenQA::Schema::Result::Jobs::INCOMPLETE_RESULTS(), OpenQA::Schema::Result::Jobs::FAILED());
```
Comment (foursixnine) [inline t/13-joblocks.t]: How about combining both together instead? as NOT_OK_RESULTS?

File: lib/OpenQA/Resource/Locks.pm (hunk 1 above)
Comment (mudler) [reply lib/OpenQA/Resource/Locks.pm]: You mean a comment line? or..?

**Comment by coolo** [comment]
shouldn't this be documented in one way or the other before this is merged?

File: lib/OpenQA/Resource/Locks.pm (hunk 1 above)
Comment (foursixnine) [reply lib/OpenQA/Resource/Locks.pm]: a blank line between 140 and 141, I just get a bit lost on where the condition ends... :)

Comment (mudler) [reply lib/OpenQA/Resource/Locks.pm]: done! 👍

File: t/13-joblocks.t (hunk 2 above)
Comment (mudler) [reply t/13-joblocks.t]: 👍

**Comment by mudler** [comment]
@coolo : can't actually locate any documentation for barriers, where should i extend it?

**Comment by coolo** [comment]
And that doesn't worry you? :)

**Pull request merged**